### ISSUE TITLE
linchpin destroy doesn't remove the libvirt networks

### ISSUE THREAD

**Issue by mcornea** [CentOS-PaaS-SIG/linchpin#940]
Steps to reproduce:

(.venv) [root@sealusa2 rhhi]# cat PinFile 
---
libvirt-network:
  topology:
    topology_name: "libvirt_network"
    resource_groups:
      - resource_group_name: "libvirt_network"
        resource_group_type: "libvirt"
        resource_definitions:
          - name: external
            role: libvirt_network
            uri: qemu:///system
            ip: 192.168.123.1
            dhcp_start: 192.168.123.100
            dhcp_end: 192.168.123.150
            domain: redhat.local
            bridge: "external"

(.venv) [root@sealusa2 rhhi]# linchpin up                                                                                                                                                                                                     
 [WARNING]: Unable to parse /root/linchpin/rhhi/localhost as an inventory source

 [WARNING]: No inventory was parsed, only implicit localhost is available

Action 'up' on Target 'libvirt-network' is complete

ID: 12
Action: up

Target                  Run ID  uHash   Exit Code 
-------------------------------------------------
libvirt-network             13  a48ffc          0

(.venv) [root@sealusa2 rhhi]# linchpin destroy
 [WARNING]: Unable to parse /root/linchpin/rhhi/localhost as an inventory source

 [WARNING]: No inventory was parsed, only implicit localhost is available

Action 'destroy' on Target 'libvirt-network' is complete

ID: 13
Action: destroy

Target                  Run ID  uHash   Exit Code 
-------------------------------------------------
libvirt-network             14  a48ffc          0

We can see that the external bridge/interface are still present. 

(.venv) [root@sealusa2 rhhi]# brctl show external
bridge name     bridge id               STP enabled     interfaces
external                8000.525400fae28a       yes             external-nic
(.venv) [root@sealusa2 rhhi]# ip a s dev external
104: external: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN group default qlen 1000
    link/ether 52:54:00:fa:e2:8a brd ff:ff:ff:ff:ff:ff
    inet 192.168.123.1/24 brd 192.168.123.255 scope global external
       valid_lft forever preferred_lft forever
(.venv) [root@sealusa2 rhhi]# virsh net-list
 Name                 State      Autostart     Persistent
----------------------------------------------------------
 default              active     yes           yes
 external             active     no            yes

**Comment by mcornea** [comment]
linchpin version 1.7.1

**Comment by samvarankashyap** [comment]
@mcornea Yes it does not. It was decided not to teardown the networks by default while desigining the networks inorder not to mess up the environment accidentally . However , This can be changed if its required.

**Comment by mcornea** [comment]
@samvarankashyap Is there any option to teardown the networks during destroy? Perhaps we could make it optional and keep the current behavior the default.

**Comment by samvarankashyap** [comment]
@mcornea I can have an option like "delete_on_destroy: true/false" within the topology would that work for you ?

**Comment by mcornea** [comment]
@samvarankashyap sounds great, thanks!
